test/distributed/fsdp/test_fsdp_use_orig_params.py

```diff
@@ -0,0 +1,875 @@
+                    fsdp1.named_parameters(),
+                    fsdp2.named_parameters(),
+                ):
+                    self.assertEqual(n1, n2)
+                    torch.testing.assert_close(p1, p2)
+
+    def _get_fsdp_parity_subtest_config(self):
+        return {
+            "sharding_strategy": [
+                ShardingStrategy.NO_SHARD,
+                ShardingStrategy.SHARD_GRAD_OP,
+                ShardingStrategy.FULL_SHARD,
+            ],
+        }
+
+    @skip_if_lt_x_gpu(2)
+    @parametrize("offload_params", [False, True])
+    def test_multiple_forward(self, offload_params: bool):
+        """
+        Tests that ``use_orig_params=True`` has parity with ``False`` when
+        running multiple forward passes before a backward pass.
+        """
+        cpu_offload = CPUOffload(offload_params=offload_params)
+        self.run_subtests(
+            self._get_fsdp_parity_subtest_config(),
+            self._test_multiple_forward,
+            cpu_offload=cpu_offload,
+        )
+
+    @skip_if_lt_x_gpu(2)
+    def _test_multiple_forward(
+        self,
+        sharding_strategy: ShardingStrategy,
+        cpu_offload: CPUOffload,
+    ):
+        (
+            fsdp_model,
+            optim,
+            fsdp_model_orig_params,
+            optim_orig_params,
+        ) = self._get_fsdp_models_and_optims(sharding_strategy, cpu_offload)
+        device = torch.device("cuda")
+        for i in range(3):
+            optim.zero_grad()
+            optim_orig_params.zero_grad()
+
+            inp1 = fsdp_model.get_input(device)
+            loss1 = fsdp_model(*inp1)
+            loss_orig_params1 = fsdp_model_orig_params(*inp1)
+            self.assertEqual(loss1, loss_orig_params1)
+
+            inp2 = fsdp_model.get_input(device)
+            loss2 = fsdp_model(*inp2)
+            loss_orig_params2 = fsdp_model_orig_params(*inp2)
+            self.assertEqual(loss2, loss_orig_params2)
+
+            loss = (loss1 + loss2).sum()
+            loss_orig_params = (loss_orig_params1 + loss_orig_params2).sum()
+            fsdp_model.run_backward(loss)
+            fsdp_model_orig_params.run_backward(loss_orig_params)
+            optim.step()
+            optim_orig_params.step()
+        self._check_fsdp_parameter_parity(fsdp_model, fsdp_model_orig_params)
+
+    @skip_if_lt_x_gpu(2)
+    @parametrize("offload_params", [False, True])
+    def test_forward_summon_forward(self, offload_params: bool):
+        """
+        Tests that ``use_orig_params=True`` has parity with ``False`` when
+        running a forward pass, :meth:`summon_full_params()`, and another
+        forward pass before a backward pass.
+        """
+        cpu_offload = CPUOffload(offload_params=offload_params)
+        self.run_subtests(
+            self._get_fsdp_parity_subtest_config(),
+            self._test_forward_summon_forward,
+            cpu_offload=cpu_offload,
+        )
+
+    def _test_forward_summon_forward(
```

Comment:
_test_forward_summon_in_between_forward?

Reply:
Changed to `test_summon_between_two_forwards()` and `_test_summon_between_two_forwards()`.